transformers/events/median_drop/transformer.go

```diff
@@ -0,0 +1,65 @@
+package median_drop
+
+import (
+	"strconv"
+
+	"github.com/lib/pq"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/shared"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/shared/constants"
+	"github.com/makerdao/vulcanizedb/libraries/shared/factories/event"
+	"github.com/makerdao/vulcanizedb/pkg/core"
+	"github.com/makerdao/vulcanizedb/pkg/datastore/postgres"
+)
+
+type Transformer struct{}
+
+func (Transformer) ToModels(_ string, logs []core.EventLog, db *postgres.DB) ([]event.InsertionModel, error) {
+	var (
+		models []event.InsertionModel
+	)
+	for _, log := range logs {
+		err := shared.VerifyLog(log.Log, shared.FourTopicsRequired, shared.LogDataRequired)
+		if err != nil {
+			return nil, err
+		}
+
+		contractAddressID, contractAddressErr := shared.GetOrCreateAddress(log.Log.Address.String(), db)
+		if contractAddressErr != nil {
+			return nil, shared.ErrCouldNotCreateFK(contractAddressErr)
+		}
+
+		msgSender := log.Log.Topics[1].Hex()
+		msgSenderAddressID, msgSenderAddressErr := shared.GetOrCreateAddress(msgSender, db)
+		if msgSenderAddressErr != nil {
+			return nil, shared.ErrCouldNotCreateFK(msgSenderAddressErr)
+		}
+
+		aLength, aLengthErr := strconv.ParseUint(log.Log.Topics[3].Hex(), 0, 64)
```

Comment:
Not sure, but I think that when a `LogNote` event gets an array, `Topics[2]` is the size of 1 element of the array in bytes. For all the events I've seen, it's always just the number 20 (padded out to fit the topic ofc), which makes sense since the only arrays I've seen as arguments are arrays of addresses, which are 20 bytes long.

Reply:
Ah makes sense - I guess we probably don't need that data (and could potentially grab it off the log if we wanted it in the future?). Cool that the log contains array length + item length - I imagine that could prove useful for general purpose helpers built around parsing arrays in events